### DISCUSSION TITLE
ComboBox - SelectedValue

### DISCUSSION THREAD

**Discussion by marcosgerene** [AvaloniaUI/Avalonia#13248] [Q&A] [answered]
Hi folks,

I have this ComboBox.

```
<ComboBox SelectedValueBinding="{Binding Id}" DisplayMemberBinding="{Binding Nome}"
ItemsSource="{Binding Pecas}" SelectedValue="{Binding ItemAlterando.Peca.Id}"  />
```

When window is created,  ItemAlterando.Peca.Id is 0 (zero) and there is no 0 on Pecas (Id).

So I receive a error opening the window:
```
Message: System.NullReferenceException: 'Object reference not set to an instance of an object.'
StackTrace;    at CompiledAvaloniaXaml.XamlIlHelpers.Scantimburgo.Calculos.Core.Model.Peca,Scantimburgo.Calculos.Core.Id!Setter(Object , Object )
```

When ItemAlterando.Peca is null it works. When I remove SelectedValue from ComboBox it works. 

How to make it work when ItemAlterando.Peca.Id does not fit on Pecas Id's?

**Comment by timunie** [comment]
ItemsSource should be the first property. If that doesn't help, please file a minimal sample

**Reply by workgroupengineering** [reply]
I @timunie i have same issue. 
It not only depende from ItemsSource position. The workaround ti is Set bind property as Nullable and set initial value in Loaded events.

The workaround is to set the property binded with SelectedValue as Nullable and set the initial value in the Loaded event.

I will open an issue with this and other issues associated with `SelectingItemsControl`

[Sandox.zip](https://github.com/AvaloniaUI/Avalonia/files/13280342/Sandox.zip)

**Reply by timunie** [reply]
❤️ thx.

**Reply by gurocky** [reply]
Anynews about the selectedValue binding issue?  controls under nested itemcontrols are really hard to reload during loaded event?

**Reply by marcosgerene** [reply]
@timunie 
I don't need this anymore (by now), but... it always interesting to keep informed... so... Is there any news? 😄

**Reply by timunie** [reply]
Maybe you can try 11.1 beta to check if anything changed during binding refractor. Could be but I didn't check it recently.

**Reply by marcosgerene** [reply]
How come you don't know everything? (kidding haha)

I will try it =)

**Reply by marcosgerene** [reply, marked as answer]
@timunie 

It works on 11.1-beta2!

My test: [test.zip](https://github.com/user-attachments/files/15598106/test.zip)

![test](https://github.com/AvaloniaUI/Avalonia/assets/15945003/8248a352-38cc-417a-9b3d-1d24cf1607a3)

**Reply by gurocky** [reply]
SelectedValue with SelectedValueBinding Could not be Initialize Correctly event in 11.1.0 beta2  -- "The workaround is to set the property binded with SelectedValue as Nullable and set the initial value in the Loaded event."

**Reply by gurocky** [reply]
Sorry, I also could not found the desktop version of test.

**Comment by workgroupengineering** [comment]
@marcosgerene  Can you try the PR #13544?

**Comment by marcosgerene** [comment]
@workgroupengineering 

I just changed the component for an TextBox and a Searching window 'cause my project was behind schedule, sorry =(
